libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -2572,6 +2580,18 @@ bool AP_Mission::calc_rewind_pos(Mission_Command& rewind_cmd)
     return true;
 }
 
+//get next navigation command which also has valid location
+bool AP_Mission::get_next_nav_cmd_with_loc(Mission_Command& cmd) {
+    for (uint8_t i = _nav_cmd.index; i < (unsigned)_cmd_total; i++) {
```

Comment:
This needs the reset check to see if it should start at 1.

Reply:
Can you point me to how I can do this check?

Reply:
https://github.com/ArduPilot/ardupilot/pull/21294#discussion_r932061517